### DISCUSSION TITLE
Now always get `400 Bad Request` in HTTP responses instead of usual Fasify validator detailed error object

### DISCUSSION THREAD

**Discussion by WhereJuly** [fastify/fastify#5566] [Q&A] [answered]
It used to return the like object in the response.body as a result of the failed request validation :
``` json
{
  "statusCode": 400,
  "error": "Bad Request",
  "message": "body should have required property 'name'"
}
```
Now the response body is empty, only status code is `400` and status message is `Bad Request`.

I use `setValidatorCompiler` now with Ajv, tried to switch back to basic `Fastify({ ajv: { customOptions: ..} });`. That did not change the thing. Could not find anything resembling the way out in docs.

Would much appreciate a heads up towards a solution.

**Comment by Eomm** [comment]
Could you add a [Minimal, Reproducible Example](https://stackoverflow.com/help/minimal-reproducible-example)?

Without it, we are unable to help you.

**Reply by WhereJuly** [reply]
@Eomm thanks for the reaction. With all the respect, unfortunately no.

No need for code explorations and debugging.

I appreciate quick heads up from somebody here that had the same issue or is experinced enough to see potential culprit by the given description.

**Reply by Eomm** [reply]
I wrote that component, am I experienced enough?

You are asking for consultancy and a trial and error approach. 

The causes are:
- custom error handlers
- routes schema response

Good luck

**Reply by WhereJuly** [reply]
> I wrote that component,

Thanks a lot for contributing to such an overall lovely framework.

I definitely support the formal approach with reproduction. But in my logic this is Discussion board, not the issues list, where the formality is a necessity to prevent mess. 

The project I faced the issue in has grown pretty big. To reproduce an error could be a project in itself. For that I just do not have time. The behaviour in question is not critical, I leave it to the point where it is justified to spend time for.

> You are asking for consultancy and a trial and error approach.

I clearly stated three times that I wasn't :)

Thanks for the hints. Be well.

**Reply by WhereJuly** [reply, marked as answer]
For consistency sake would describe the solution.
This is on Fastify v4.28.x

The culprit is in a [response serialization schema](https://fastify.dev/docs/v4.28.x/Reference/Validation-and-Serialization/#serialization) that you provide (or do not provide) for the route.

If your response schema is just `{}` Fastify will serialize it as such, an empty object which you get in the response JSON. Irrespectively of what content you actually put there in your route handler by `reply.code(200).send(`{content: "value"}`)`.

If you do not define the route response schema you get your reply content `as is`  e.g. as in above example.

If your response schema and the response content on intersection create some object, you get that object.

And finally to get at least Fastify error format either 1) assign the route the error response schema defined as described in the [docs](https://fastify.dev/docs/v4.28.x/Reference/Reply/#errors) or 2) do not define error response schema at all.

**Comment by jsumners** [comment]
Closing due to lack of cooperation.
